### PR TITLE
Maintain selected type for each collection, and pick next type when disabling the active type

### PR DESCRIPTION
**Pull request by foxtree**
also:
1.) pick the next _enabled_ type when disabling the active/selected type
2.) pick the type that is being toggled via the filter-drawer, if all filters were disabled

### CODE REVIEW

File: src/redux/reducers/index.js [hunk 1]
```diff
@@ -100,9 +100,22 @@ export const collectionsReducer = (state = preloadCollections, action) => {
...
+            .map((type, index, array) => ({
+              type,
+              next: array[(index === array.length - 1) ? 0 : index + 1],
+            }))
+            .find(({ type }) => type === resourceType);
+          if (thisType?.next) {
+            collection.selectedResourceType = thisType.next;
+          }
+        }
+        collection.resourceTypeFilters[resourceType] = nextValue; // eslint-disable-line no-param-reassign, max-len
```
Comment (foxtree) [inline src/redux/reducers/index.js:118]: ^ this is in the last commit, which could be dropped

File: src/redux/reducers/index.js [hunk 2]
```diff
@@ -71,6 +71,14 @@ const preloadCollections = {
   [defaultCollection.id]: defaultCollection,
 };
 
+const getNextEnabledType = (resourceType, enabledTypes) => enabledTypes
+  .map((type, index, array) => ({
+    type,
+    next: array[(index === array.length - 1) ? 0 : index + 1],
```
Comment (foxtree) [inline src/redux/reducers/index.js:77]: alternatively, the "next" type could be the next previous one that is enabled. (leftward)

...unless it's the left-most that is being disabled, then pick the next to the right.

File: src/redux/reducers/index.js [hunk 3]
```diff
@@ -38,8 +38,8 @@ const createCollection = (label = 'Untitled Collection') => {
     created: timeCreated,
     lastUpdated: timeCreated,
     label,
-    selectedResourceType: null,
-    resourceTypeFilters: Object.keys(PLURAL_RESOURCE_TYPES)
+    selectedResourceType: TYPES_SORTED_BY_LABEL[0],
+    resourceTypeFilters: TYPES_SORTED_BY_LABEL
```
Comment (gilhsu) [inline src/redux/reducers/index.js:42]: I realize that Master does the same, but it seems pretty fake the way we establish the initial `resourceTypeFilters` through a list of constants.  It should be dynamic and built based on the records returned for the patient. 

Is it confirmed that all the test patients have at least 1 resource for each resourceType?

Comment (foxtree) [reply src/redux/reducers/index.js:42]: The type filters don't respond to what records are available. This seems like something that could be adapted, here,  _after_ that functionality is implemented.

**Review by gilhsu — APPROVED** [review]

**Pull request merged**